### ISSUE TITLE
Required arguments are scattered under "optional arguments"

### ISSUE THREAD

**Issue by kwinkunks** [equinor/komodo#279]
`kmd --help` lists arguments like so (my annotations on the right):

```
positional arguments:
  pkgs
  repo

optional arguments:
  -h, --help
  --prefix PREFIX, -p PREFIX           <-- actually required ------
  --release RELEASE, -r RELEASE        <-- actually required ------
  --tmp TMP, -t TMP
  --cache CACHE, -c CACHE
  --jobs JOBS, -j JOBS
  --download, -d
  --build, -b
  --install, -i
  --dry-run, -n
  --cmake CMAKE
  --pip PIP
  --virtualenv VIRTUALENV
  --pyver PYVER
  --sudo
  --workspace WORKSPACE
  --extra-data-dirs EXTRA_DATA_DIRS [EXTRA_DATA_DIRS ...]
  --postinst POSTINST, -P POSTINST
  --locations-config LOCATIONS_CONFIG  <-- actually required ------
  --renamer RENAMER, -R RENAMER
```

This is because, by default, `argparse` puts non-positional arguments under "optional arguments", in the order in which they are added to the parser.

It's not all bad news: they are shown without square brackets in the `usage` section, and they are listed in the error message you get if you leave them out.

I suggest creating an "argument group", per [the `argpase` documentation](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_argument_group) and making a group called "required named arguments", for example:

```
required_args = parser.add_argument_group('required named arguments')
required_args.add_argument("--prefix", "-p", type=str, required=True)
required_args.add_argument("--release", "-r", type=str, required=True)
required_args.add_argument(
    "--locations-config",
    type=str,
    required=True,
    help="Path to locations.yml, a map of location to a server.",
)
```

**Comment by kwinkunks** [comment]
Worked on this with #282.

**Comment by kwinkunks** [comment]
Resolved by 80d99efeed95f0ffe37ba08448a4896c8df93dde
